### ISSUE TITLE
@vite blade directive generates the wrong links when Laravel is behind a reverse proxy

### ISSUE THREAD

**Issue by torenware** [laravel/framework#43540]
- Laravel Version: 

- PHP Version: 9.23.0
- Database Driver & Version:  n/a

### Description:

When running the Laravel 9.23 sample app behind DDEV 1.19, Laravel is running behind a reverse proxy which terminates SSL and assumes a fixed hostname (e.g., my-laravel-app.ddev.site).  When developing JavaScript with the Vite plugin, the `@vite()` blade directive will resolve to the following:

In blade:
```blade
   @vite(['resources/css/app.css', 'resources/js/app.js'])
```

In HTML as resolved:
```html
<script type="module" src="[http://[::]:5173/@vite/client](view-source:http://[::]:5173/@vite/client)"></script><link rel="stylesheet" href="[http://[::]:5173/resources/css/app.css](view-source:http://[::]:5173/resources/css/app.css)" /><script type="module" src="[http://[::]:5173/resources/js/app.js](view-source:http://[::]:5173/resources/js/app.js)"></script>
```

Note that Laravel is writing this out relative to HTTP and localhost (IPv6) which is not where the reverse proxy is exposing these links. The links would need to point to something under `https://my-laravel.ddev.site:5173` to be correctly routed through the reverse proxy.

### Steps To Reproduce:

[A user has posted a repo to use with ddev](https://github.com/Ellarddekoeijer/my-laravel-app).  Add the `@vite()` directive to welcome.blade.php to get this working.  In addition:

1. Run `ddev get torenware/viteserve`
2. Edit the generated .ddev/.env  file so that `VITE_PROJECT_DIR=.` so that the ddev vite server finds vite.config.js
3. Run `ddev vite-serve` to start the ddev vite server.
4. Run `ddev restart`
5. Add the `@vite` directive to welcome.blade.php
6. Browse to https://my-laravel-app.ddev.site/
7. Look at the generated HTML in the browser, and note that the `@vite` directive resolves to the wrong protocol and domain.

**Comment by torenware** [comment]
Main issue here is: where and how can a user specify the right path to the Vite served assets when behind a reverse proxy, as is the case with DDEV.

**Comment by driesvints** [comment]
Seems like Jess has already answered your questions here: https://github.com/laravel/vite-plugin/issues/83

**Comment by TheEvilCoder42** [comment]
@driesvints Hello, I don't see a reasonable answer in this issue.

My site is proxyied, but also available on another url, which results in CORS & CSP issues.
My current workaround is to omit the url.
To achieve that, I set `ASSET_URL=.` in `.env` and in the `package.json` I set the script command to `build: ASSET_URL='' vite build`

I'm not sure if this would also solve the DDEV issue, but I propose an option in the `.env` called `ASSET_RELATIVE` or something, which omits the url in the `@vite` directive.
That way the url won't make problems like it currently does and is simply relative and it doesn't have to be clever about the url (KISS).
